### PR TITLE
Feature/export data

### PR DESCRIPTION
**Pull request by venus-sherathiya**
Finally, I think it works !! Lets review it when we meet

### CODE REVIEW

File: src/bento.py [hunk 1]
```diff
@@ -542,14 +576,14 @@ def quit(self, event):
             time.sleep(3./30.)  # wait for threads to shut down
             QApplication.instance().quit()
 
-    def newVideoWidget(self, video_path: str, start_time: Timecode, forcePixmapMode: bool) -> VideoFrame:
-        video = VideoFrame(self)
+    def newVideoWidget(self, video_path: str, id: int, start_time: Timecode, forcePixmapMode: bool) -> VideoFrame:
+        video = VideoFrame(self, id)
```
Comment (venus-sherathiya) [inline src/bento.py:580]: get rid of id from everywhere

File: src/bento.py [hunk 2]
```diff
@@ -668,7 +710,7 @@ def loadTrial(self, videos, annotation, loadPose, loadNeural, loadAudio):
             # We give native videos priority by artificially pushing their start times 6 hours earlier
             pairs = [[video_data.start_time, video_data] for video_data in videos]
             for item in pairs:
-                if VideoFrame(self).supported_by_native_player(item[1].file_path):
+                if VideoFrame(self, 0).supported_by_native_player(item[1].file_path):
```
Comment (venus-sherathiya) [inline src/bento.py:713]: get rid of id from everywhere

File: src/bento.py [hunk 3]
```diff
@@ -680,8 +722,7 @@ def loadTrial(self, videos, annotation, loadPose, loadNeural, loadAudio):
                 else:
                     path = video_data.file_path
                 # force pixmap mode if we already are using a native player as a time source
-                widget = self.newVideoWidget(fix_path(path), video_data.start_time, bool(timeSource))
-                self.video_widgets.append(widget)
+                widget = self.newVideoWidget(fix_path(path), ix, video_data.start_time, bool(timeSource))
```
Comment (venus-sherathiya) [inline src/bento.py:725]: get rid of id/ix

File: src/dataExporter.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+"""
+
+from pynwb import NWBFile
+
+class DataExporter:
+    """
+    Data export base class
+    """
+
+    def __init__(self, id: int):
```
Comment (venus-sherathiya) [inline src/dataExporter.py:15]: get rid of id

File: src/video/videoWindow.py [hunk 5]
```diff
@@ -5,16 +5,20 @@
...
+class VideoFrame(QFrame, DataExporter):
 
     openReader = Signal(str)
     quitting = Signal()
 
-    def __init__(self, bento):
-        super().__init__()
+    def __init__(self, bento, id):
+        QFrame.__init__(self)
+        DataExporter.__init__(self, id)
```
Comment (venus-sherathiya) [inline src/video/videoWindow.py:20]: get rid of id

File: src/video/videoWindow.py [hunk 6]
```diff
@@ -49,17 +53,17 @@ def mouseReleaseEvent(self, event):
 
     def supported_by_native_player(self, fn: str) -> bool:
         ext = os.path.basename(fn).rsplit('.',1)[-1].lower()
-        if ext in VideoSceneNative(self.bento, self.bento.current_time()).supportedFormats():
+        if ext in VideoSceneNative(self.bento, self.id, self.bento.current_time()).supportedFormats():
```
Comment (venus-sherathiya) [inline src/video/videoWindow.py:56]: get rid of id

File: src/video/videoWindow.py [hunk 7]
```diff
@@ -49,17 +53,17 @@ def mouseReleaseEvent(self, event):
...
             return True
-        if not ext in VideoScenePixmap(self.bento, self.bento.current_time()).supportedFormats():
+        if not ext in VideoScenePixmap(self.bento, self.id, self.bento.current_time()).supportedFormats():
```
Comment (venus-sherathiya) [inline src/video/videoWindow.py:58]: get rid of id

File: src/video/videoWindow.py [hunk 8]
```diff
@@ -49,17 +53,17 @@ def mouseReleaseEvent(self, event):
...
             raise Exception(f"video format {ext} not supported.")
         return False
 
     def load_video(self, fn: str, start_time: Timecode, forcePixmapMode: bool):
         if not forcePixmapMode and self.supported_by_native_player(fn):
-            self.scene = VideoSceneNative(self.bento, start_time)
+            self.scene = VideoSceneNative(self.bento, self.id, start_time)
```
Comment (venus-sherathiya) [inline src/video/videoWindow.py:64]: get rid of id

File: src/video/videoWindow.py [hunk 9]
```diff
@@ -49,17 +53,17 @@ def mouseReleaseEvent(self, event):
...
         else:
-            self.scene = VideoScenePixmap(self.bento, start_time)
+            self.scene = VideoScenePixmap(self.bento, self.id, start_time)
```
Comment (venus-sherathiya) [inline src/video/videoWindow.py:66]: get eid of id

File: src/video/videoScene.py [hunk 10]
```diff
@@ -24,8 +26,9 @@ class VideoSceneAbstractBase(QGraphicsScene):
     def supportedFormats() -> List:
         raise NotImplementedError("Derived class needs to override this method")
 
-    def __init__(self, bento: QObject, start_time: Timecode, parent: QObject=None):
-        super().__init__(parent)
+    def __init__(self, bento: QObject, id: int, start_time: Timecode, parent: QObject=None):
+        QGraphicsScene.__init__(self, parent)
+        DataExporter.__init__(self, id)
```
Comment (venus-sherathiya) [inline src/video/videoScene.py:31]: get rid id

File: src/plugin/plugin_pose_DeepLabCut.py [hunk 11]
```diff
@@ -45,8 +50,8 @@ def _validateFileCSV(self, parent_widget, file_path: str) -> bool:
             reader = csv.reader(csvFile)
             header1 = next(reader)
             if (not isinstance(header1[1], str) or
-                header1[1] != 'bodyparts' or
-                (len(header1) - 2) % 3 != 0):   # "", "bodyparts", 3 x <bodypart_name>, ...
+                header1[1] != 'body_parts' or
```
Comment (venus-sherathiya) [inline src/plugin/plugin_pose_DeepLabCut.py:53]: get rid of underscore

File: src/plugin/plugin_pose_DeepLabCut.py [hunk 12]
```diff
@@ -143,33 +151,77 @@ def _loadPoses_h5(self, parent_widget, path: str):
...
-                this_frame_points = []
-                for point_ix in range(num_pts):
-                    pt_x_ix = (vals_per_pt * point_ix) + 2
-                    pt_y_ix = (vals_per_pt * point_ix) + 3
-                    this_frame_points.append(QPointF(float(row[pt_x_ix]), float(row[pt_y_ix])))
-                self.frame_points.append(this_frame_points)
-            self.num_frames = len(self.frame_points)
-
+    def _loadPoses_csv(self, parent_widget, path: str, video_path: str):
+        df = pd.read_csv(path, header=[0,1], index_col=0)
```
Comment (venus-sherathiya) [inline src/plugin/plugin_pose_DeepLabCut.py:155]: change header=[0,1,2]

File: src/plugin/plugin_pose_MARS.py [hunk 13]
```diff
@@ -92,6 +104,101 @@ def loadPoses(self, parent_widget: QWidget, path: str):
...
+                    data = self.keypoints[:,mouse_ix,:,3],
+                    reference_frame = '(0,0,0) corresponds to ...',
+                    timestamps = np.arange(self.num_frames, dtype=float),
+                    confidence = self.confidence[:,mouse_ix,3]
+                )
+            )
+            pose_estimation_series.append(
+                PoseEstimationSeries(
+                    name = 'left hip',
+                    description = 'Pose keypoint placed aroud left hip',
```
Comment (venus-sherathiya) [inline src/plugin/plugin_pose_MARS.py:155]: check description everywhere (aroud to around)
reference_frame to top left corner

File: src/widgets/neuralWidget.py [hunk 14]
```diff
@@ -386,4 +387,15 @@ def showAnnotations(self, enabled):
         if isinstance(self.annotations, QGraphicsItem):
             self.annotations.setVisible(enabled)
         if isinstance(self.heatmap, QGraphicsItem):
-            self.heatmap.setOpacity(0.5 if enabled else 1.)
\ No newline at end of file
+            self.heatmap.setOpacity(0.5 if enabled else 1.)
+    
+    def exportToNWBFile(self, nwbFile: NWBFile):
+        neuralData = TimeSeries(name=f"neural_data",
+                                data = self.data,
```
[the remark this answers is not in the document]
Comment (annkennedy) [reply src/widgets/neuralWidget.py:394]: fixed in the BTA branch

**Pull request merged**